src/Visualization/Python/GenerateXdmf.py

```diff
@@ -214,7 +214,13 @@ def data_item(data_type):
                         "AttributeType=\"Vector\" Center=\"Node\">\n" %
                         (vector))
                     xdmf_output += data_item_vec
-                    for index in ["_x", "_y", "_z"]:
+
+                    index_list = []
+                    if dimensionality == 2:
+                        index_list = ["_x", "_y"]
+                    elif dimensionality == 3:
+                        index_list = ["_x", "_y", "_z"]
```

Comment:
This looks good, but what do you think of writing it in fewer lines like this?
```
index_list = ["_x", "_y"]
if dimensionality == 3:
    index_list.append("_z")
```
(I haven't tested this works, but something like this should...)

Reply:
Worked out successfully (at least for advection 2D data) -- do you have any 3D volume data in mind for a quick double-check test?

Reply:
I think the NewtonianEuler 3D Riemann problem should be fairly easy to use as a check: I think you can add some options to the input file to observe the density (a scalar) and the velocity (a vector) in 3D and that should be a sufficient check.

Reply:
Check completed